engine/util/rng.hpp

```diff
@@ -65,33 +68,45 @@ struct basic_rng_t
   /// Uniform distribution in the range [min..max)
   double range( double min, double max );
 
-  /// Uniform distribution in the range [min..max)
-  template <typename T, typename = std::enable_if_t<std::is_integral<T>::value>>
-  T range( T min, T max ) {
-    return static_cast<T>(range(static_cast<double>(min), static_cast<double>(max)));
-  }
-
-  /// Uniform distribution in the range [0..max)
-  template <typename T, typename = std::enable_if_t<std::is_integral<T>::value>>
-  T range( T max ) {
-    return range<T>( T{}, max );
-  }
```

Comment:
I think we do still need these range templates, or rather we still need the second one as well as this version with a floor to correct the bug with `min < 0` to replace the first one. In addition to fixing that bug, not having these templates to do the integer to double static casts results in a significant number of new warnings with some compilers (e.g., MSVC).
```cpp
  /// Uniform distribution in the range [min..max)
  template <typename T, typename = std::enable_if_t<std::is_integral<T>::value>>
  T range( T min, T max ) {
    return static_cast<T>( std::floor( range( static_cast<double>( min ), static_cast<double>( max ) ) ) );
  }
```